File: src/main/java/com/github/ccguyka/showupdates/consumer/ReadUpdatesFile.java
package com.github.ccguyka.showupdates.consumer;

import java.io.File;
import java.io.IOException;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.SerializationFeature;
import com.github.ccguyka.showupdates.objects.ProjectUpdates;

public class ReadUpdatesFile {

    private final ObjectMapper mapper;

    public ReadUpdatesFile() {

        this.mapper = new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);
    }

    public ProjectUpdates readFromFile(File updatesFile) throws IOException {
        return mapper.readValue(updatesFile, ProjectUpdates.class);
    }
}

File: src/main/kotlin/com/github/ccguyka/showupdates/consumer/ReadUpdatesFile.kt
package com.github.ccguyka.showupdates.consumer

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.databind.SerializationFeature
import com.github.ccguyka.showupdates.objects.ProjectUpdates
import java.io.File
import java.io.IOException

class ReadUpdatesFile {

    private val mapper: ObjectMapper = ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT)

    @Throws(IOException::class)
    fun readFromFile(updatesFile: File?): ProjectUpdates {
        return mapper.readValue(updatesFile, ProjectUpdates::class.java)
    }
}
